internal/common/objects/BaseBiker.go

```diff
@@ -50,21 +53,55 @@ type BaseBiker struct {
 	energyLevel                      float64 // float between 0 and 1
 	points                           int
 	alive                            bool
+	megaBikeId                       uuid.UUID               
+}
+
+func (bb *BaseBiker) GetLocation(gameState utils.IGameState) utils.Coordinates {
+	gs := gameState.GetGameState()
+	return gs.MegaBikes[bb.megaBikeId].coordinates
+}
```

Comment:
General comment regarding implementation:

It may be nice to have the server pass in the gamestate to agents each turn, i.e., with `agent.UpdateGameState(GameState)` - this gives server total control over the gamestate it exposes. It also means that an agent can call `GetLocation()` without any arguments, as the necessary data exists within the agent struct. This would then allow agents to call `bb.gamestate`, say.

Again, this is entirely up to you as a cohort, but this way seems more 'pattern', and may be worth a discussion